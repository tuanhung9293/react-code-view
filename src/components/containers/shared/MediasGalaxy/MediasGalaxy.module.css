.nothing {
    width: 100%;
    text-align: center;
    font-size: 20px;
    padding: 20px 0;
}

.info {
    margin-top: 20px;
}