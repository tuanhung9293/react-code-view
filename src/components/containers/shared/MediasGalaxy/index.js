import React from 'react'
import {connect} from 'react-redux'
import {Columns, Column} from 'components/ui/grid'
import NasaMediaItem from 'components/ui/NasaMediaItem'
import {fetchMediaItemCollection} from 'state/actions/mediaAction'
import styles from './MediasGalaxy.module.css'

class MediasGalaxy extends React.Component {
  render() {
    const {galaxyData, source, actionComponent, fetchCollectionUrl, total, searchParams} = this.props
    return (
      <React.Fragment>
        <div className={styles.info}>
          {total > 0 &&
            <div>
              <span><b>{galaxyData.length}</b> loaded results of <b>{total}</b></span>
              {source === 'medias' && <span> for <b>"{searchParams.q}"</b></span>}
            </div>
          }
        </div>
        <Columns>
          {galaxyData.map((item, key) =>
            <Column key={key} span={3} breakPoints={[{maxWidth: 1200, span: 4}, {maxWidth: 768, span: 6}, {maxWidth: 576, span: 12}]}>
              <NasaMediaItem
                itemData={item}
                source={source}
                onFetchCollectionUrl={() => fetchCollectionUrl(key, item.mediaType, item.href)}
                actionComponent={actionComponent} />
            </Column>
          )}
          {galaxyData.length === 0 && <div className={styles.nothing}>No item to show...</div>}
        </Columns>
      </React.Fragment>
    )
  }
}

export default connect(
  ({medias}) => ({searchParams: medias.params}),
  (dispatch) => ({
    fetchCollectionUrl: (index, type, url) => dispatch(fetchMediaItemCollection(index, type, url))
  })
)(MediasGalaxy)
