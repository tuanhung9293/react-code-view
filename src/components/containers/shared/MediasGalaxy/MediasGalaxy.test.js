import React from 'react'
import {mount, shallow} from 'enzyme'
import {Provider} from 'react-redux'
import {MemoryRouter} from 'react-router-dom'
import MediasGalaxy from './'

import configureStore from 'redux-mock-store'
const mockStore = configureStore()

let store = mockStore({
  medias: {
    data: [{}, {}, {}],
    totalHits: 1002,
    params: {
      q: 'dada',
      mediaType: 'video'
    },
    loading: true,
    loadingMore: false
  },
  ui: {windowSize: {width: 615, height: 953}}
})
store.dispatch = () => {}

function createMockComponent() {
  return jest.fn().mockImplementation(props => null)
}

describe('renders', () => {
  it('renders MediasGalaxy connected', () => {
    expect(shallow(
      <Provider store={store}>
        <MediasGalaxy
          galaxyData={[]}
          source="medias"
          actionComponent={createMockComponent()}
          total={100}
        />
      </Provider>
    )).toMatchSnapshot()
  })

  it('renders MediasGalaxy Component empty element', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MediasGalaxy
          galaxyData={[]}
          source="medias"
          actionComponent={createMockComponent()}
          total={0}
        />
      </Provider>
    )
    expect(wrapper.find(MediasGalaxy)).toMatchSnapshot()
  })

  it('renders MediasGalaxy Component filled elements', () => {
    jest.mock('../../../ui/NasaMediaItem', () => ()=> <div id="mockNasaMediaItem">
      NasaMediaItem
    </div>)

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{pathname: '/', key: 'testKey'} ]}>
          <MediasGalaxy
            galaxyData={[{}, {}, {}]}
            source="collection"
            actionComponent={createMockComponent()}
            total={3}
          />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(MediasGalaxy)).toMatchSnapshot()
  })
})

