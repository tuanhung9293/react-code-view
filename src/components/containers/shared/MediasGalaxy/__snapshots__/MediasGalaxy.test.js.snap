// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders MediasGalaxy Component empty element 1`] = `
<Connect(MediasGalaxy)
  actionComponent={[MockFunction]}
  galaxyData={Array []}
  source="medias"
  total={0}
>
  <MediasGalaxy
    actionComponent={[MockFunction]}
    fetchCollectionUrl={[Function]}
    galaxyData={Array []}
    searchParams={
      Object {
        "mediaType": "video",
        "q": "dada",
      }
    }
    source="medias"
    total={0}
  >
    <div
      className="info"
    />
    <Columns>
      <div
        className="base"
      >
        <div
          className="nothing"
        >
          No item to show...
        </div>
      </div>
    </Columns>
  </MediasGalaxy>
</Connect(MediasGalaxy)>
`;

exports[`renders renders MediasGalaxy Component filled elements 1`] = `
<Connect(MediasGalaxy)
  actionComponent={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "itemData": Object {},
          },
          Object {},
        ],
        Array [
          Object {
            "itemData": Object {},
          },
          Object {},
        ],
        Array [
          Object {
            "itemData": Object {},
          },
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": null,
        },
        Object {
          "isThrow": false,
          "value": null,
        },
        Object {
          "isThrow": false,
          "value": null,
        },
      ],
    }
  }
  galaxyData={
    Array [
      Object {},
      Object {},
      Object {},
    ]
  }
  source="collection"
  total={3}
>
  <MediasGalaxy
    actionComponent={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "itemData": Object {},
            },
            Object {},
          ],
          Array [
            Object {
              "itemData": Object {},
            },
            Object {},
          ],
          Array [
            Object {
              "itemData": Object {},
            },
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": null,
          },
          Object {
            "isThrow": false,
            "value": null,
          },
          Object {
            "isThrow": false,
            "value": null,
          },
        ],
      }
    }
    fetchCollectionUrl={[Function]}
    galaxyData={
      Array [
        Object {},
        Object {},
        Object {},
      ]
    }
    searchParams={
      Object {
        "mediaType": "video",
        "q": "dada",
      }
    }
    source="collection"
    total={3}
  >
    <div
      className="info"
    >
      <div>
        <span>
          <b>
            3
          </b>
           loaded results of 
          <b>
            3
          </b>
        </span>
      </div>
    </div>
    <Columns>
      <div
        className="base"
      >
        <Connect(Column)
          breakPoints={
            Array [
              Object {
                "maxWidth": 1200,
                "span": 4,
              },
              Object {
                "maxWidth": 768,
                "span": 6,
              },
              Object {
                "maxWidth": 576,
                "span": 12,
              },
            ]
          }
          key="0"
          span={3}
        >
          <Column
            breakPoints={
              Array [
                Object {
                  "maxWidth": 1200,
                  "span": 4,
                },
                Object {
                  "maxWidth": 768,
                  "span": 6,
                },
                Object {
                  "maxWidth": 576,
                  "span": 12,
                },
              ]
            }
            dispatch={[Function]}
            span={3}
            windowWidth={615}
          >
            <div
              className="base grid"
              style={
                Object {
                  "WebkitBoxFlex": 0,
                  "flexBasis": "50%",
                  "flexGrow": 0,
                  "maxWidth": "50%",
                  "msFlexPositive": 0,
                  "msFlexPreferredSize": "50%",
                }
              }
            >
              <NasaMediaItem
                actionComponent={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                    ],
                  }
                }
                itemData={Object {}}
                onFetchCollectionUrl={[Function]}
                source="collection"
              >
                <div
                  className="wrapper"
                >
                  <div
                    className="inner"
                  >
                    <MediaThumb
                      itemData={Object {}}
                      onFetchCollectionUrl={[Function]}
                      source="collection"
                    >
                      <div
                        className="wrapper"
                      >
                        <img
                          className="thumb"
                        />
                        <div
                          className="playButton"
                        >
                          <PlayButton
                            to="/play-media/collection/undefined"
                          >
                            <Link
                              replace={false}
                              to="/play-media/collection/undefined"
                            >
                              <a
                                href="/play-media/collection/undefined"
                                onClick={[Function]}
                              >
                                <div
                                  className="wrapper"
                                >
                                  <svg
                                    viewBox="0 0 26 26"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <polygon
                                      className="svg"
                                      points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
                                    />
                                    <path
                                      className="svg"
                                      d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z"
                                    />
                                  </svg>
                                </div>
                              </a>
                            </Link>
                          </PlayButton>
                        </div>
                      </div>
                    </MediaThumb>
                    <div
                      className="mediaType"
                    >
                      <a />
                    </div>
                    <div
                      className="info"
                    >
                      <h4
                        className="title"
                      >
                        <a />
                      </h4>
                      <div
                        className="meta"
                      >
                        <div
                          className="postauthor"
                        >
                          By 
                          <a>
                            unknown
                          </a>
                        </div>
                        <div>
                            /  
                          <img
                            height="10px"
                            src="test-file-stub"
                          />
                           
                          Invalid date
                        </div>
                      </div>
                      <div
                        className="description"
                      />
                    </div>
                    <div
                      className="addButton"
                    >
                      <mockConstructor
                        itemData={Object {}}
                      />
                    </div>
                  </div>
                </div>
              </NasaMediaItem>
            </div>
          </Column>
        </Connect(Column)>
        <Connect(Column)
          breakPoints={
            Array [
              Object {
                "maxWidth": 1200,
                "span": 4,
              },
              Object {
                "maxWidth": 768,
                "span": 6,
              },
              Object {
                "maxWidth": 576,
                "span": 12,
              },
            ]
          }
          key="1"
          span={3}
        >
          <Column
            breakPoints={
              Array [
                Object {
                  "maxWidth": 1200,
                  "span": 4,
                },
                Object {
                  "maxWidth": 768,
                  "span": 6,
                },
                Object {
                  "maxWidth": 576,
                  "span": 12,
                },
              ]
            }
            dispatch={[Function]}
            span={3}
            windowWidth={615}
          >
            <div
              className="base grid"
              style={
                Object {
                  "WebkitBoxFlex": 0,
                  "flexBasis": "50%",
                  "flexGrow": 0,
                  "maxWidth": "50%",
                  "msFlexPositive": 0,
                  "msFlexPreferredSize": "50%",
                }
              }
            >
              <NasaMediaItem
                actionComponent={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                    ],
                  }
                }
                itemData={Object {}}
                onFetchCollectionUrl={[Function]}
                source="collection"
              >
                <div
                  className="wrapper"
                >
                  <div
                    className="inner"
                  >
                    <MediaThumb
                      itemData={Object {}}
                      onFetchCollectionUrl={[Function]}
                      source="collection"
                    >
                      <div
                        className="wrapper"
                      >
                        <img
                          className="thumb"
                        />
                        <div
                          className="playButton"
                        >
                          <PlayButton
                            to="/play-media/collection/undefined"
                          >
                            <Link
                              replace={false}
                              to="/play-media/collection/undefined"
                            >
                              <a
                                href="/play-media/collection/undefined"
                                onClick={[Function]}
                              >
                                <div
                                  className="wrapper"
                                >
                                  <svg
                                    viewBox="0 0 26 26"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <polygon
                                      className="svg"
                                      points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
                                    />
                                    <path
                                      className="svg"
                                      d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z"
                                    />
                                  </svg>
                                </div>
                              </a>
                            </Link>
                          </PlayButton>
                        </div>
                      </div>
                    </MediaThumb>
                    <div
                      className="mediaType"
                    >
                      <a />
                    </div>
                    <div
                      className="info"
                    >
                      <h4
                        className="title"
                      >
                        <a />
                      </h4>
                      <div
                        className="meta"
                      >
                        <div
                          className="postauthor"
                        >
                          By 
                          <a>
                            unknown
                          </a>
                        </div>
                        <div>
                            /  
                          <img
                            height="10px"
                            src="test-file-stub"
                          />
                           
                          Invalid date
                        </div>
                      </div>
                      <div
                        className="description"
                      />
                    </div>
                    <div
                      className="addButton"
                    >
                      <mockConstructor
                        itemData={Object {}}
                      />
                    </div>
                  </div>
                </div>
              </NasaMediaItem>
            </div>
          </Column>
        </Connect(Column)>
        <Connect(Column)
          breakPoints={
            Array [
              Object {
                "maxWidth": 1200,
                "span": 4,
              },
              Object {
                "maxWidth": 768,
                "span": 6,
              },
              Object {
                "maxWidth": 576,
                "span": 12,
              },
            ]
          }
          key="2"
          span={3}
        >
          <Column
            breakPoints={
              Array [
                Object {
                  "maxWidth": 1200,
                  "span": 4,
                },
                Object {
                  "maxWidth": 768,
                  "span": 6,
                },
                Object {
                  "maxWidth": 576,
                  "span": 12,
                },
              ]
            }
            dispatch={[Function]}
            span={3}
            windowWidth={615}
          >
            <div
              className="base grid"
              style={
                Object {
                  "WebkitBoxFlex": 0,
                  "flexBasis": "50%",
                  "flexGrow": 0,
                  "maxWidth": "50%",
                  "msFlexPositive": 0,
                  "msFlexPreferredSize": "50%",
                }
              }
            >
              <NasaMediaItem
                actionComponent={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                      Array [
                        Object {
                          "itemData": Object {},
                        },
                        Object {},
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                      Object {
                        "isThrow": false,
                        "value": null,
                      },
                    ],
                  }
                }
                itemData={Object {}}
                onFetchCollectionUrl={[Function]}
                source="collection"
              >
                <div
                  className="wrapper"
                >
                  <div
                    className="inner"
                  >
                    <MediaThumb
                      itemData={Object {}}
                      onFetchCollectionUrl={[Function]}
                      source="collection"
                    >
                      <div
                        className="wrapper"
                      >
                        <img
                          className="thumb"
                        />
                        <div
                          className="playButton"
                        >
                          <PlayButton
                            to="/play-media/collection/undefined"
                          >
                            <Link
                              replace={false}
                              to="/play-media/collection/undefined"
                            >
                              <a
                                href="/play-media/collection/undefined"
                                onClick={[Function]}
                              >
                                <div
                                  className="wrapper"
                                >
                                  <svg
                                    viewBox="0 0 26 26"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <polygon
                                      className="svg"
                                      points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
                                    />
                                    <path
                                      className="svg"
                                      d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z"
                                    />
                                  </svg>
                                </div>
                              </a>
                            </Link>
                          </PlayButton>
                        </div>
                      </div>
                    </MediaThumb>
                    <div
                      className="mediaType"
                    >
                      <a />
                    </div>
                    <div
                      className="info"
                    >
                      <h4
                        className="title"
                      >
                        <a />
                      </h4>
                      <div
                        className="meta"
                      >
                        <div
                          className="postauthor"
                        >
                          By 
                          <a>
                            unknown
                          </a>
                        </div>
                        <div>
                            /  
                          <img
                            height="10px"
                            src="test-file-stub"
                          />
                           
                          Invalid date
                        </div>
                      </div>
                      <div
                        className="description"
                      />
                    </div>
                    <div
                      className="addButton"
                    >
                      <mockConstructor
                        itemData={Object {}}
                      />
                    </div>
                  </div>
                </div>
              </NasaMediaItem>
            </div>
          </Column>
        </Connect(Column)>
      </div>
    </Columns>
  </MediasGalaxy>
</Connect(MediasGalaxy)>
`;

exports[`renders renders MediasGalaxy connected 1`] = `
<Connect(MediasGalaxy)
  actionComponent={[MockFunction]}
  galaxyData={Array []}
  source="medias"
  total={100}
/>
`;
